packages/ruleset/src/functions/schema-naming-convention.js

```diff
@@ -0,0 +1,474 @@
+    if (
+      source.toLowerCase().endsWith('yaml') ||
+      source.toLowerCase().endsWith('yml') ||
+      source.toLowerCase().endsWith('json') ||
+      source === 'root' // This is necessary for running the tests
+    ) {
+      const refMap = nodes[source].refMap;
+
+      // Each resolved path to a schema is stored with a path to its referenced
+      // schema in 'components'. Sub-schemas within components also have their
+      // paths stored with the path to the schema they reference. This gathers
+      // the paths, transforming them from Spectral's internal format, and maps
+      // them to the name of the schema they reference.
+      Object.keys(refMap).forEach(pathWithRef => {
+        const path = pathWithRef
+          .split('/')
+          .map(p => decodeURIComponent(p.replaceAll('~1', '/')))
+          .join('.')
+          .slice(2);
+        resultMap[path] = refMap[pathWithRef].slice(2).replaceAll('/', '.');
+      });
+    }
+  });
+
+  return resultMap;
+}
+
+/**
+ * Takes an unresolved path to a schema and un-resolves it to the format in which
+ * it will be stored in the graph nodes map by Spectral. The nodes provide a
+ * map from resolved path locations to the locations in 'components' they
+ * reference (if there is a reference at that location). Each path will be
+ * resolved to their nearest parent. For example if a request body schema
+ * references a schema in components, there will be an entry mapping the path
+ * to the request body schema to the schema path in components - but if that
+ * referenced schema has a property defined by a reference to another schema,
+ * there will not be an entry in the map including the path leading through
+ * the request body to the sub-property, there will be an entry mapping from
+ * the location of the schema in components. Here are example entries:
+ * - paths./v1/things.post.requestBody.content.application/json.schema:
+ *     components.schemas.ThingPrototype
+ * - components.schemas.ThingPrototype.properties.data:
+ *     components.schemas.DataObject
+ *
+ * The purpose of this function is to take the fully resolved path (in the
+ * example above, something like 'paths./v1/things.post.requestBody.content.
+ * application/json.schema.properties.data') and un-resolve enough that it
+ * will match the format Spectral uses in nodes (e.g. 'components.schemas.
+ * ThingPrototype.properties.data').
+ *
+ * @param {string} path - the resolved JSON path to a schema, as a dot-separated string
+ * @param {object} pathToReferencesMap - the graph nodes map from Spectral, which has
+ *                                       been sanitzed for our purposes. it maps referenced
+ *                                       schema locations to their reference location.
+ * @returns {string} - the name of the referenced schema at a given path, or undefined
+ */
+function getSchemaNameAtPath(path, pathToReferencesMap) {
+  if (!path && typeof path !== 'string') {
```

Comment:
Should this actually be:
```suggestion
  if (!path || typeof path !== 'string') {
```
?

Reply:
Yes 🙂